### PR TITLE
[i18n-ar] Translated file : `docs/source/ar/llm_tutorial_optimization.md` into Arabic

### PR DESCRIPTION
**Pull request by AhmedAlmaghz**
## What does this PR do?
Translated the `docs/source/ar/llm_tutorial_optimization.md` file of the documentation to `Arabic العربية`. I use transformers for my side-project and wanted to contribute. As a beginner, this seemed appropriate.

### Some questions:

- I can't access the pr documentation endpoint for `Arabic العربية`. Is there a missing step?
- I left all the other files alone after copying the English docs. Is this acceptable for future updates or should I add contents to the `_toctree.yml` only when translation is complete? Should I delete all other files for a cleaner review?
- In `Arabic`, people use both `Transformer` and `المحولات` to describe the model. However, as this is more of a product of HuggingFace I opted not to translate it. May you please let me know which you would prefer?

Thank you in advance for your review.

Part of #32435

### Before submitting

- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests), Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/) ?Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

 ### Who can review?
 @stevhliu , @MKhalusova , @abodacs, may you please review this PR?

### CODE REVIEW

File: docs/source/ar/llm_tutorial_optimization.md [hunk 1]
```diff
@@ -0,0 +1,634 @@
...
+
+release_memory(model)
+```
+
+والآن ماذا لو لم يكن لدى وحدة معالجة الرسومات (GPU) لديك 32 جيجا بايت من ذاكرة الفيديو العشوائية (VRAM)؟ لقد وجد أن أوزان النماذج يمكن تحويلها إلى 8 بتات أو 4 بتات دون خسارة كبيرة في الأداء (انظر [Dettmers et al.](https://arxiv.org/abs/2208.07339)).
+يمكن تحويل النموذج إلى 3 بتات أو 2 بتات مع فقدان مقبول في الأداء كما هو موضح في ورقة [GPTQ](https://arxiv.org/abs/2210.17323) 🤯.
+
+دون الدخول في الكثير من التفاصيل، تهدف مخططات التحويل إلى تقليل دقة الأوزان مع محاولة الحفاظ على دقة نتائج النموذج كما هي (*أي* أقرب ما يمكن إلى bfloat16).
+لاحظ أن التحويل يعمل بشكل خاص جيدًا لتوليد النص حيث كل ما نهتم به هو اختيار *مجموعة الرموز الأكثر احتمالًا التالية* ولا نهتم حقًا بالقيم الدقيقة لتوزيع الرمز التالي *logit*.
+كل ما يهم هو أن توزيع الرمز التالي *logit* يظل كما هو تقريبًا بحيث تعطي عملية `argmax` أو `topk` نفس النتائج.
```
Comment (abodacs) [inline docs/source/ar/llm_tutorial_optimization.md:160]:
```suggestion
دون الدخول في الكثير من التفاصيل، تهدف مخططات التكميم إلى تخفيض دقة الأوزان مع محاولة الحفاظ على دقة نتائج النموذج كما هي (*أي* أقرب ما يمكن إلى bfloat16).
لاحظ أن التكميم يعمل بشكل خاص جيدًا لتوليد النص حيث كل ما نهتم به هو اختيار *مجموعة الرموز الأكثر احتمالًا التالية* ولا نهتم حقًا بالقيم الدقيقة لتوزيع الرمز التالي *logit*.
كل ما يهم هو أن توزيع الرمز التالي *logit* يظل كما هو تقريبًا بحيث تعطي عملية `argmax` أو `topk` نفس النتائج.
```

File: docs/source/ar/llm_tutorial_optimization.md [hunk 2]
```diff
@@ -0,0 +1,634 @@
...
+
+هناك تقنيات تحويل مختلفة، والتي لن نناقشها بالتفصيل هنا، ولكن بشكل عام، تعمل جميع تقنيات التحويل كما يلي:
```
Comment (abodacs) [inline docs/source/ar/llm_tutorial_optimization.md:162]:
```suggestion
هناك عدة تقنيات للتكميم، والتي لن نناقشها بالتفصيل هنا، ولكن بشكل عام، تعمل جميع تقنيات التكميم كما يلي:
```

File: docs/source/ar/llm_tutorial_optimization.md [hunk 3]
```diff
@@ -0,0 +1,634 @@
...
+
+-   1.  قم بتحويل جميع الأوزان إلى الدقة المستهدفة
+-   2.  قم بتحميل الأوزان المحولة، ومرر تسلسل الإدخال من المتجهات بتنسيق bfloat16
+-   3.  قم بتحويل الأوزان ديناميكيًا إلى bfloat1
```
Comment (abodacs) [inline docs/source/ar/llm_tutorial_optimization.md:166]:
```suggestion
-   1.  تكميم جميع الأوزان إلى الدقة المستهدفة
-   2.  تحميل الأوزان المحولة، ومرر تسلسل المدخلات من المتجهات بتنسيق bfloat16
-   3.  قم بتحويل الأوزان ديناميكيًا إلى bfloat1  لإجراء الحسابات مع متجهات المدخلات بدقة `bfloat16`
```

File: docs/source/ar/llm_tutorial_optimization.md [hunk 4]
```diff
@@ -0,0 +1,634 @@
...
+لم أترجم النصوص الخاصة ولا الأكواد البرمجية ولا الروابط ولا رموز HTML و CSS، كما طلبت.
```
Comment (abodacs) [inline docs/source/ar/llm_tutorial_optimization.md:167]:
```suggestion
```

File: docs/source/ar/llm_tutorial_optimization.md [hunk 5]
```diff
@@ -0,0 +1,634 @@
...
+
+---
+
+يعد \\( \mathbf{X} = (\mathbf{x}_1, ... \mathbf{x}_{N}) \\) بالتالي تسلسل الإدخال إلى طبقة الاهتمام. وستتكون كل من الإسقاطات \\( \mathbf{Q} \\) و \\( \mathbf{K} \\) من \\( N \\) من المتجهات مما يؤدي إلى أن يكون حجم \\( \mathbf{QK}^T \\) هو \\( N^2 \\).
+
+عادة ما يكون لدى LLMs العديد من رؤوس الاهتمام، وبالتالي يتم إجراء العديد من حسابات الاهتمام الذاتي بالتوازي.
+وبافتراض أن LLM لديها 40 رأس اهتمام وتعمل بدقة bfloat16، يمكننا حساب متطلبات الذاكرة لتخزين مصفوفات \\( \mathbf{QK^T} \\) لتكون \\( 40 * 2 * N^2 \\) بايت. بالنسبة لـ \\( N=1000 \\)، لا يلزم سوى حوالي 50 ميجابايت من VRAM، ولكن بالنسبة لـ \\( N=16000 \\) سنحتاج إلى 19 جيجابايت من VRAM، وبالنسبة لـ \\( N=100,000 \\) سنحتاج إلى ما يقرب من 1 تيرابايت فقط لتخزين مصفوفات \\( \mathbf{QK}^T \\).
+
+باختصار، سرعان ما يصبح خوارزمية الاهتمام الذاتي الافتراضية مكلفة للغاية من حيث الذاكرة بالنسبة لسياقات الإدخال الكبيرة.
```
Comment (abodacs) [inline docs/source/ar/llm_tutorial_optimization.md:176]:
```suggestion
باختصار، سرعان ما يصبح خوارزمية الانتباه الذاتي الافتراضية مكلفة للغاية من حيث الذاكرة بالنسبة لسياقات الإدخال الكبيرة.
```

File: docs/source/ar/llm_tutorial_optimization.md [hunk 6]
```diff
@@ -0,0 +1,634 @@
...
+
+> ومع ذلك، فإن الاهتمام الفلاشي أسرع بكثير في الاستدلال مقارنة بالاهتمام الافتراضي الذي يأتي من قدرته على تقليل الطلبات على ذاكرة GPU الأبطأ ذات النطاق الترددي العالي (VRAM)، والتركيز بدلاً من ذلك على ذاكرة SRAM الأسرع الموجودة على الشريحة.
+
+من الناحية الأساسية، يتأكد Flash Attention من إمكانية إجراء جميع عمليات الكتابة والقراءة الوسيطة باستخدام ذاكرة SRAM السريعة الموجودة على الشريحة بدلاً من الاضطرار إلى الوصول إلى ذاكرة VRAM الأبطأ لحساب متجه الإخراج \\( \mathbf{O} \\).
+
+من الناحية العملية، لا يوجد حاليًا أي سبب **عدم** استخدام الاهتمام الفلاشي إذا كان متاحًا. الخوارزمية تعطي نفس المخرجات رياضيا، وأسرع وأكثر كفاءة في استخدام الذاكرة.
+
+لنلقِ نظرة على مثال عملي.
+
+لنلقِ نظرة على مثال عملي.
```
Comment (abodacs) [inline docs/source/ar/llm_tutorial_optimization.md:204]:
```suggestion
```

**Review by abodacs — CHANGES_REQUESTED** [review]
السﻻم عليكم
@AhmedAlmaghz 
اعتقد ان هنالك بعض الفقرات الناقصة فى الترحمه

**Comment by AhmedAlmaghz** [comment]
مرحبا أخي عبدالله @abodacs 
اشكرك على المراجعة وأعذرني على الإنقطاع اليومين السابقين لقد  كنت مريضاً
سوف أراجع الملف و اتحقق من الناقص

يمكنك البدء بالقسم التالي :

# DEVELOPER GUIDES

- [ ]  Use fast tokenizers from 🤗 Tokenizers #33034
- [ ]  Run inference with multilingual models #33048
- [ ]  Use model-specific APIs #33030
- [ ]  Share a custom model #33031
- [ ]  Templates for chat models #33026
- [ ]  Trainer #33080
- [ ]  Run training on Amazon SageMaker #33071
- [ ]  Export to ONNX #33072
- [ ]  Export to TFLite #33077
- [ ]  Export to TorchScript #33079
- [ ]  Benchmarks #33023
- [ ]  Notebooks with examples #33049
- [ ]  Community resources #33027
- [ ]  Troubleshoot #33017
- [ ]  Interoperability with GGUF files #33037

**Comment by abodacs** [comment]
@AhmedAlmaghz 
شفاك الله وعافاك أخى

سأحاول أراجع ﻷن السبوع القادم عندى امتحانات

**Comment by AhmedAlmaghz** [comment]
حفظك الله ورعاك أخي عبدالله @abodacs 
بالتوفيق إن شاء الله في إمتحاناتك
اعمل approving  لكل PRs الجاهزة حتى يتم دمجها
لقد راجعت هذا الملف وأكملت الناقص أشكرك  يمكنك مراجعته

تحياتي وتقديري الكبيرين لشخصكم الكريم

**Review by abodacs — APPROVED** [review]
@AhmedAlmaghz 
برجاء معالجة التعارض

**Pull request closed without merging**